community/licensecheck-config/src/test/java/org/neo4j/licensecheck/LicenseYearTest.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) 2002-2017 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.licensecheck;
+
+import org.junit.Test;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.net.URL;
+import java.util.Calendar;
+import java.util.Scanner;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.fail;
+
+public class LicenseYearTest
+{
+
+    private static final String NOTICE_GPL_FILE = "notice-gpl-prefix.txt";
+    private static final String NOTICE_AGPL_FILE = "notice-agpl-prefix.txt";
+
+    @Test
+    public void testNoticePrefixYearGPL() throws FileNotFoundException
+    {
+        URL resource = getClass().getClassLoader().getResource( NOTICE_GPL_FILE );
+        File gplFile = new File( resource.getFile() );
+
+        checkYearInFile(gplFile);
+    }
+
+    @Test
+    public void testNoticePrefixYearAGPL() throws FileNotFoundException
+    {
+        URL resource = getClass().getClassLoader().getResource( NOTICE_AGPL_FILE );
+        File gplFile = new File( resource.getFile() );
+
+        checkYearInFile( gplFile );
+    }
```

Comment:
I would think that any version that we build has to have this field updated. If not, what is the point of having the year there at all? This means that at new year's, we'll update this branch's and all subsequent branches' `NOTICE` files, and then this wouldn't be a problem. What do you think?

Of course, you could make the argument that in a future where `3.3` enters the archive, then we'd stop updating and then any build past a new year's would fail, but I think at that point _very_ few builds will be made. I'd not want to spend too much time catering for that use case (this stuff with build parameters is a bit foreign to me).